libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/SfapApiGenerations.kt

```diff
@@ -0,0 +1,79 @@
+package com.salesforce.androidsdk.auth
+
+import com.salesforce.androidsdk.auth.SfapApiGenerationsResponseBody.Companion.fromJson
+import com.salesforce.androidsdk.rest.RestClient
+import com.salesforce.androidsdk.rest.RestRequest
+import com.salesforce.androidsdk.rest.RestRequest.RestMethod.POST
+import kotlinx.serialization.json.Json.Default.encodeToString
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
+
+/**
+ * Provides generated text from the `sfap_api` "generations" endpoint.
+ *
+ * See https://developer.salesforce.com/docs/einstein/genai/guide/access-models-api-with-rest.html
+ *
+ * @param apiHostName The Salesforce `sfap_api` hostname
+ * @param modelName The model name to request generations from.  For possible
+ * values, see https://developer.salesforce.com/docs/einstein/genai/guide/api-names.html
+ * @param restClient The REST client to use
+ */
+class SfapApiGenerations(
```

Comment:
I wonder if we should have request/response helper objects, but let the app do the sending and decide if it wants the raw response or the parsed response.

We did something like that for [composite requests](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/CompositeRequest.java).

Reply:
We're not using JSON.org here and Kotlin Serialization took its place.  I think that's a win, so perhaps something like https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2644/commits/90c9f2eb48086bf1cd60dfd2ba11381785f052a6 gives the app a convenience to generate JSON from the request and also to retrieve the endpoint's source JSON from the response.  How's that look?